app/controllers/tasks_controller.rb

```diff
@@ -21,6 +26,28 @@ def create
     redirect_to "/"
   end
 
+  def edit
+    id = params[:id]
+    @task = Task.find(id)
+  end
+
+  def complete
+    complete = Task.find(params[:id])
+    complete.completed_at = Time.now
+    complete.save
+    redirect_to "/tasks"
+  end
+
+  def update
+    id = params[:id]
+    @task = Task.find(id)
+    @task.update(
```

Comment:
Because you have written the `task_params` method and it uses `params.permit` you can actually simplify this line to: `@task.update(task_params[:task])` and that will automatically grab both the `name` and `description` values from the params hash.